File: test/Models/pheno_pkg/src/java/Ismomentregistredzc_39.java
import  java.io.*;
import  java.util.*;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import javafx.util.*;
import java.time.LocalDateTime;
public class Ismomentregistredzc_39
{
    
    public Ismomentregistredzc_39() { }
    public void  Calculate_ismomentregistredzc_39(PhenologyState s, PhenologyState s1, PhenologyRate r, PhenologyAuxiliary a)
    {
        //- Name: IsMomentRegistredZC_39 -Version: 1.0, -Time step: 1
        //- Description:
    //            * Title: Is FlagLeafLiguleJustVisible Model
    //            * Author: Pierre Martre
    //            * Reference: Modeling development phase in the 
    //                Wheat Simulation Model SiriusQuality.
    //                See documentation at http://www1.clermont.inra.fr/siriusquality/?page_id=427
    //            * Institution: INRA Montpellier
    //            * Abstract: if FlagLeafLiguleJustVisible is already Registred 
        //- inputs:
    //            * name: calendarMoments_t1
    //                          ** description : List containing appearance of each stage at previous time
    //                          ** variablecategory : state
    //                          ** datatype : STRINGLIST
    //                          ** default : ['Sowing']
    //                          ** unit : 
    //                          ** inputtype : variable
        //- outputs:
    //            * name: isMomentRegistredZC_39
    //                          ** description :  if Flag leaf ligule has already appeared 
    //                          ** variablecategory : state
    //                          ** datatype : INT
    //                          ** min : 0
    //                          ** max : 1
    //                          ** unit : 
        List<String> calendarMoments_t1 = s1.getcalendarMoments();
        int isMomentRegistredZC_39;
        isMomentRegistredZC_39 = calendarMoments_t1.contains("FlagLeafLiguleJustVisible") ? 1 : 0;
        s.setisMomentRegistredZC_39(isMomentRegistredZC_39);
    }
}

File: src/pycropml/transpiler/antlr_py/tests/examples/SiriusComponent/phenology/original/src/py/original/ismomentregistredzc_39.py
# coding: utf8
from copy import copy
from array import array
from math import *
from typing import *
from datetime import datetime

import numpy

#%%CyML Model Begin%%
def model_ismomentregistredzc_39(calendarMoments_t1:List[str]):
    """
     - Name: IsMomentRegistredZC_39 -Version: 001, -Time step: 1
     - Description:
                 * Title: IsMomentRegistredZC_39 model
                 * Authors: Pierre Martre
                 * Reference: ('',)
                 * Institution: INRA Montpellier
                 * ExtendedDescription: if FlagLeafLiguleJustVisible is already Registred 
                 * ShortDescription: None
     - inputs:
                 * name: calendarMoments_t1
                               ** description : List containing appearance of each stage
                               ** inputtype : variable
                               ** variablecategory : state
                               ** datatype : STRINGLIST
                               ** max : 
                               ** min : 
                               ** default : 
                               ** unit : 
     - outputs:
                 * name: isMomentRegistredZC_39
                               ** description :  if Flag leaf ligule has already appeared 
                               ** datatype : INT
                               ** variablecategory : state
                               ** max : 1
                               ** min : 0
                               ** unit : 
    """

    isMomentRegistredZC_39:int
    isMomentRegistredZC_39 = 1 if "FlagLeafLiguleJustVisible" in calendarMoments_t1 else 0
    return isMomentRegistredZC_39
#%%CyML Model End%%